selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/Csharp/CSharpRenamingTest.java

```diff
@@ -0,0 +1,95 @@
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.client.TestCommandServiceClient;
+import org.eclipse.che.selenium.core.client.TestProjectServiceClient;
+import org.eclipse.che.selenium.core.workspace.InjectTestWorkspace;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.core.workspace.WorkspaceTemplate;
+import org.eclipse.che.selenium.pageobject.CodenvyEditor;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.eclipse.che.selenium.pageobject.intelligent.CommandsPalette;
+import org.openqa.selenium.TimeoutException;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class CSharpRenamingTest {
+  private static final String PROJECT_NAME =
+      NameGenerator.generate(CSharpRenamingTest.class.getSimpleName(), 4);
+
+  private static final String PATH_TO_DOTNET_FILE = PROJECT_NAME + "/Hello.cs";
+
+  @InjectTestWorkspace(template = WorkspaceTemplate.UBUNTU_LSP)
+  private TestWorkspace workspace;
+
+  @Inject private Ide ide;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private Loader loader;
+  @Inject private CodenvyEditor editor;
+  @Inject private Menu menu;
+  @Inject private Wizard wizard;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private TestCommandServiceClient testCommandServiceClient;
+  @Inject private CommandsPalette commandsPalette;
+  @Inject private Consoles consoles;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = getClass().getResource("/projects/CsharpHelloWorld");
+    testProjectServiceClient.importProject(
+        workspace.getId(), Paths.get(resource.toURI()), PROJECT_NAME, DOT_NET);
+    ide.open(workspace);
+    projectExplorer.openItemByPath(PROJECT_NAME);
+    projectExplorer.openItemByPath(PATH_TO_DOTNET_FILE);
+    consoles.waitExpectedTextIntoConsole(FINISH_LANGUAGE_SERVER_INITIALIZATION_MESSAGE);
+  }
+
+  @Test
+  public void checkRenaming() {
+    String newClassName = "HelloWorld";
+    String textFragmentAfterRenaming = "public class HelloWorld";
+    projectExplorer.openItemByPath(PROJECT_NAME);
+    consoles.selectProcessByTabName("dev-machine");
+    editor.goToCursorPositionVisible(18, 18);
+    menu.runCommand(ASSISTANT, REFACTORING, LS_RENAME);
+    editor.doRenamingByLanguageServerField(newClassName);
+
+    try {
+      editor.waitTextIntoEditor(textFragmentAfterRenaming);
+    } catch (TimeoutException ex) {
+      // remove try-catch block after issue has been resolved
+      fail("Known issue https://github.com/eclipse/che/issues/10180", ex);
```

Comment:
Is it permanent failure? If so, it's better to express it in error message like the follow:
```
Known permanent failure https://github.com/eclipse/che/issues/10180
```